packages/shell-api/src/helpers.ts

```diff
@@ -120,3 +122,322 @@ export function processDigestPassword(username, passwordDigestor, command): any
   }
   return { digestPassword: true };
 }
+
+
+/**
+ * Return an object which will become a ShardingStatusResult
+ * @param mongo
+ * @param configDB
+ * @param verbose
+ */
+export async function getPrintableShardStatus(mongo: Mongo, verbose: boolean): Promise<any> {
+  const result = {} as any; // use array to maintain order
+
+  // configDB is a DB object that contains the sharding metadata of interest.
+  // Defaults to the db named "config" on the current connection.
+  const configDB = mongo.getDB('config');
+  const mongosColl = configDB.getCollection('mongos');
+  const versionColl = configDB.getCollection('version');
+  const shardsColl = configDB.getCollection('shards');
+  const chunksColl = configDB.getCollection('chunks');
+  const settingsColl = configDB.getCollection('settings');
+  const changelogColl = configDB.getCollection('changelog');
+
+  const version = await versionColl.findOne();
+  if (version === null) {
+    throw new MongoshInvalidInputError('This db does not have sharding enabled.' +
+      ' be sure you are connecting to a mongos from the shell and not to a mongod.');
+  }
+
+  result.shardingVersion = await versionColl.findOne();
```

Comment:
This could be

```suggestion
  result.shardingVersion = version;
```

right?

Reply:
Yes, thanks!